lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -1463,19 +1454,68 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :allowed_from_server => true,
           :description => 'Distributed tracing lets you see the path that a request takes through your distributed system. Enabling distributed tracing changes the behavior of some New Relic features, so carefully consult the [transition guide](/docs/transition-guide-distributed-tracing) before you enable this feature.'
         },
+        :'distributed_tracing.sampler.root' => {
+          :default => 'default',
+          :public => true,
+          :type => String,
+          :allowed_from_server => false,
+          :allowlist => %w[default adaptive always_on always_off trace_id_ratio_based],
+          :transform => SamplerConfigValidator.validate_sampler_strategy_with_ratio(
+            :'distributed_tracing.sampler.root',
+            :'distributed_tracing.sampler.root.trace_id_ratio_based.ratio'
+          ),
+          :description => 'This setting controls the behavior of transaction sampling for transactions without a remote parent, traces that originate within this instance of the New Relic agent. Available values are `default`, `adaptive`, `always_on`, `always_off`, and `trace_id_ratio_based`. At this time `default` and `adaptive` are the same.'
+        },
         :'distributed_tracing.sampler.remote_parent_sampled' => {
           :default => 'default',
           :public => true,
           :type => String,
-          :allowed_from_server => true,
-          :description => 'This setting controls the behavior of transaction sampling when a remote parent is sampled and the trace flag is set in the traceparent. Available values are `default`, `always_on`, and `always_off`.'
+          :allowed_from_server => false,
+          :allowlist => %w[default adaptive always_on always_off trace_id_ratio_based],
+          :transform => SamplerConfigValidator.validate_sampler_strategy_with_ratio(
+            :'distributed_tracing.sampler.remote_parent_sampled',
+            :'distributed_tracing.sampler.remote_parent_sampled.trace_id_ratio_based.ratio'
+          ),
+          :description => 'This setting controls the behavior of transaction sampling when a remote parent is sampled. Available values are `default`, `adaptive`, `always_on`, `always_off`, and `trace_id_ratio_based`. At this time `default` and `adaptive` are the same.'
         },
         :'distributed_tracing.sampler.remote_parent_not_sampled' => {
           :default => 'default',
           :public => true,
           :type => String,
-          :allowed_from_server => true,
-          :description => 'This setting controls the behavior of transaction sampling when a remote parent is not sampled and the trace flag is not set in the traceparent. Available values are `default`, `always_on`, and `always_off`.'
+          :allowed_from_server => false,
+          :allowlist => %w[default adaptive always_on always_off trace_id_ratio_based],
```

Comment:
I like this allowlist feature

Reply:
me too! was happy to find it. a wonderful james feature 🍵 🦊